### DISCUSSION TITLE
How am I supposed to achieve this?

### DISCUSSION THREAD

**Discussion by joanofdart** [supabase/supabase#3405] [Questions] [answered]
Hello,

I'm not a master of PLSQL or SQL, so with that being said, I've had quite a lot of trouble getting something fairly simple to be setup, and also, something funny keeps happening and is that when I create a new function and wanna go back to edit it, the drawer is messed up.

Anyways,
I just want to be able to check, before upserting a new modification, if a given value from x column in y table, already exists.

I try creating a function,
![image](https://user-images.githubusercontent.com/43426084/135676257-69afdac5-3726-4fb9-8673-6fb2ab84b8e3.png)

tried calling it using flutter RPC... didn't work. Always got a false, tho when I execute the same code using the SQL query editor, it works properly.

I also tried using client-side-filtering, which I'm even more confused as I really don't know how to execute this, but oh well, I tried..

.ilike('username', 'blabla')

.filter('username', 'ilike', 'blabla')

So yeah... I might be really stupid but having already used 4 hours in this and getting nowhere is a pain. I know I must be doing something wrong, can someone guide me? thank you.

**Comment by silentworks** [comment]
Can you show an example of your rpc code please? also don't start function names with underscores and you don't need double underscores only single.

**Comment by awalias** [comment, marked as answer]
maybe you want to put a UNIQUE constraint on the column in question? that way postgres will automatically reject any duplicates

`ALTER TABLE tablename ADD CONSTRAINT constraintname UNIQUE (columns);` [[source](https://stackoverflow.com/questions/1194438/can-i-add-a-unique-constraint-to-a-postgresql-table-after-its-already-created)]

regarding the functions we now improved the interface so you can choose your return type, maybe this helps? 
![image](https://user-images.githubusercontent.com/458736/135823620-f4b56c4b-b7cd-4c3e-870c-c069ca91103d.png)
